File: medium/pro-divisible/Divisible.java
public final class Divisible {
    public static void main(String[] args) {
        final var scanner = new java.util.Scanner(System.in);
        final var number = scanner.nextInt();
        final var isDivisible = scanner.tokens().mapToInt(Integer::parseInt).allMatch(i -> number % i == 0); 
           
        System.out.println(isDivisible ? "divisible by all" : "not divisible by all");
    }
}

File: medium/pro-divisible/divisible.c
#include <stdio.h>

int main() {
    int n, i;
    scanf("%d\n", &n);
    
    while (scanf("%d", &i) == 1) 
        if (n % i) {
        	puts("not divisible by all");
        	return 0;
        }
    
    puts("divisible by all");
    return 0;
}
